### DISCUSSION TITLE
IPV6 connection

### DISCUSSION THREAD

**Discussion by helveden** [GeniusesOfSymfony/WebSocketBundle#445] [Q&A]
Hello my host has changed the way to set up a user program to launch the command
php bin/console gos:websocket:server

And now I would like to know if it is possible to use GOS websocket in IPV6? or how to do it?

Sincerely

**Comment by mbabker** [comment]
In theory it should, there isn't anything directly in the bundle code that would prohibit IPv6 addresses from working.

On localhost, you could try replacing `127.0.0.1` with `::1` and see what happens.

**Comment by helveden** [comment]
![image](https://user-images.githubusercontent.com/6437218/137599442-bb835500-7338-4625-9ee7-90b76ca294aa.png)

It's a good writing code for working in local ?
`ws://[::1]:1337`

**Reply by mbabker** [reply]
With the below settings, IPv6 is working for me on localhost.

```env
# .env.local
###> gos/web-socket-bundle ###
GOS_WEB_SOCKET_SERVER_IP="[::1]"
GOS_WEB_SOCKET_SERVER_PORT=1337
###< gos/web-socket-bundle ###
```

```yaml
# config/packages/gos_web_socket.yaml
gos_web_socket:
    authentication:
        providers:
            session: ~
    server:
        host: '%env(GOS_WEB_SOCKET_SERVER_IP)%'
        port: '%env(int:GOS_WEB_SOCKET_SERVER_PORT)%'
```

```twig
{# templates/websocket/index.html.twig #}
{% extends 'base.html.twig' %}

{% block title %}Hello WebsocketController!{% endblock %}

{% block javascripts %}
    <script src="{{ asset('bundles/goswebsocket/js/vendor/autobahn.min.js') }}"></script>
    <script src="{{ asset('bundles/goswebsocket/js/websocket.js') }}"></script>
{% endblock %}

{% block body %}
<style>
    .example-wrapper { margin: 1em auto; max-width: 800px; width: 95%; font: 18px/1.5 sans-serif; }
    .example-wrapper code { background: #F5F5F5; padding: 2px 6px; }
</style>

<div class="example-wrapper">
    <h1>Hello {{ controller_name }}! ✅</h1>

    Websocket Connection Status: <span id="websocket-status">Disconnected</span>
</div>
<script>
    const websocket = GosSocket.connect("ws://{{ gos_web_socket_server_host }}:{{ gos_web_socket_server_port }}");

    websocket.on("socket/connect", (session) => {
        console.log("connected");
        document.getElementById('websocket-status').innerText = 'Connected';
    });

    websocket.on("socket/disconnect", (session) => {
        console.log("disconnected");
        document.getElementById('websocket-status').innerText = 'Disconnected';
    });
</script>
{% endblock %}
```

**Reply by helveden** [reply]
It's ok in my local. I use **ws://[::1]:1337**

In my prod i use **WSS://** because my website is in https.
My pord host gave me the port 8101, but before with the functionality with the normal IP, I put the port 443 in wss and it worked. 
So far I have reproduced what I did locally with the right elements of the production, and it does not work for the moment.
